crates/core/src/metadata_store.rs

```diff
@@ -11,61 +11,150 @@
 pub mod providers;
 mod test_util;
 
-#[cfg(any(test, feature = "test-util"))]
-use crate::metadata_store::test_util::InMemoryMetadataStore;
+use std::future::Future;
+use std::sync::Arc;
+
 use async_trait::async_trait;
 use bytes::{Bytes, BytesMut};
 use bytestring::ByteString;
+use codederror::{error_code, Code, CodedError};
+use tracing::{debug, trace};
+
 use restate_types::errors::GenericError;
 use restate_types::metadata_store::keys::NODES_CONFIG_KEY;
 use restate_types::nodes_config::NodesConfiguration;
 use restate_types::retries::RetryPolicy;
 use restate_types::storage::{StorageCodec, StorageDecode, StorageEncode, StorageEncodeError};
 use restate_types::{flexbuffers_storage_encode_decode, Version, Versioned};
-use std::future::Future;
-use std::sync::Arc;
-use tracing::{debug, trace};
 
-#[derive(Debug, thiserror::Error)]
+#[cfg(any(test, feature = "test-util"))]
+use crate::metadata_store::test_util::InMemoryMetadataStore;
+
+pub const CODE_UNIMPLEMENTED: Code = error_code!(
+    "E0001",
+    help = "Unimplemented Function",
+    description =
+        "This could indicate a misconfigured client or a missing metadata-store role on node."
+);
```

Comment:
Those error definitions should go into `errors` so that they are found when generating the errors docs page (https://docs.restate.dev/references/errors).

Reply:
Didn't know about this. Thank you!